spec/controllers/answers_controller_spec.rb

```diff
@@ -11,25 +11,27 @@
 
     context 'with valid attributes' do
       it 'saves the new answer in the database' do
-        expect { post :create, question_id: question, answer: attributes_for(:answer), format: :js }
+        expect { post :create, question_id: question, answer: attributes_for(:answer), format: :json }
                .to change(question.answers, :count).by(1)
       end
 
-      it 'render create template' do
-        post :create, question_id: question, answer: attributes_for(:answer), format: :js
-        expect(response).to render_template 'create'
+      it 'renders create template' do
+        post :create, question_id: question, answer: attributes_for(:answer), format: :json
+        expect(response.header['Content-Type']).to include 'application/json'
       end
     end
 
     context 'with invalid attributes' do
       it 'does not save the answer' do
-        expect { post :create, question_id: question, answer: attributes_for(:invalid_answer), format: :js }
-               .to_not change(Answer, :count)
+        expect { post :create,
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.